### PR TITLE
Merge flutter/plugins

### PR DESCRIPTION
**Pull request by stuartmorgan**
Merges everything from the packages/ directory of flutter/plugins (which is essentially the entire remaining repository) into this repository, completing the core of the repository merge.

This is intended to be landed as a merge commit, to fully preserve commit histories, blame, etc. This will add another non-linear branch to the history, but this is consistent with how we've done other merges into this repository.

Part of https://github.com/flutter/flutter/issues/113764

### CODE REVIEW

**Comment by stuartmorgan** [comment]
Just the last handful of commits (the ones from me after the "Merge remote-tracking branch ..." commit of mine) need review, and even then it's just merging in content from the other repo so should be simple.

File: script/configs/custom_analysis.yaml
Comment (tarrinneal) [inline script/configs/custom_analysis.yaml:12]: Has this been updated, or will it be?

Comment (stuartmorgan) [reply script/configs/custom_analysis.yaml:12]: I'll update it post-merge; in the short term it'll keep working since the repo will still be there.

**Comment by stuartmorgan** [comment]
I'm not sure why url_launcher_web tests are failing; I thought I had those test configs fully aligned.

**Comment by stuartmorgan** [comment]
I checked the test configs again and I'm still not seeing any differences. I've disabled it just on stable for now; we'll still get coverage on `master` while we investigate, so it doesn't need to hold up the merge.

**Review by tarrinneal — APPROVED** [review]
As far as I can tell this all seems good. I'll cross my fingers

**Comment by stuartmorgan** [comment]
I have a few changes I'll push when the current runs all finish:
- Dart unit tests are taking a really long time now, so:
  - The Linux-host ones will be split into two shards
  - The Windows one will have `flutter_migrate` disabled because that's a very slow test that's basically doing e2e testing in a unit test.
- Windows-host unit tests will have the *_web implementation packages excluded, as they don't work. (This isn't a test configuration that existing in flutter/plugins, which I forgot, and somehow missed in my test run.)

**Comment by stuartmorgan** [comment]
Okay, everything but `Windows dart_unit_tests` passed on this run. I'm going to push the changes to both Windows and Linux unit test configs, and since those are the only things changing, call this good to land once those specific tests pass rather than waiting for all of the slowest tests.

**Review by ditman — APPROVED** [review]
Let's go!

**Comment by stuartmorgan** [comment]
Re-running with more Windows Dart unit test exclusions due to the running-unit-tests-in-Chrome issue.

**Pull request merged**
